File: Java/Missing-Number/HashSet.java
class HashSet {
    public int missingNumber(int[] nums) {
        //initialize a hash set that contains distinct elements from array nums
        Set<Integer> numSet = new HashSet<Integer>();
        for (int num : nums) {
            numSet.add(num);
        }
        //if the array nums has n elements, there must be n + 1 missing element in total
        int expectedNumsCount = nums.length + 1;
        for (int n=0; n < expectedNumsCount; n++) {
            //if n is not in numSet
            if(!numSet.contains(n)) {
                return n
            }
        }
        //if no missing number found
        return -1;
    }
}

File: Python/Missing-Number/HashSet.py
class HashSet(object):
    def missingNumber(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        #initialize a set of distinct elements in list nums
        num_set = set(nums)
        #for number in range of [0..n+1] to have space for only 1 missing number
        for number in range(len(nums) + 1):
            #if a number in range is not num_set
            if number not in num_set:
                #return the missing number
                return number
                
